app/concerns/liquid_interpolatable.rb

```diff
@@ -221,6 +221,43 @@ def logger
           Logger.new(STDERR)
         end
     end
+
+    BACKSLASH = "\\".freeze
+
+    UNESCAPE = {
+      "a" => "\a",
+      "e" => "\e",
+      "f" => "\f",
+      "n" => "\n",
+      "r" => "\r",
+      "t" => "\t",
+      "v" => "\v",
+      BACKSLASH => BACKSLASH + BACKSLASH,
+      "1" => BACKSLASH + "1",
+      "2" => BACKSLASH + "2",
+      "3" => BACKSLASH + "3",
+      "4" => BACKSLASH + "4",
+      "5" => BACKSLASH + "5",
+      "6" => BACKSLASH + "6",
+      "7" => BACKSLASH + "7",
+      "8" => BACKSLASH + "8",
+      "9" => BACKSLASH + "9",
+    }
+
+    def unescape_replacement(s)
```

Comment:
It's running a gsub across the input, replacing every matching section with the contents returned from the block.  `$1`, `$2`, etc. are the first and second match groups copied into thread-local variables.  `to_i(16)` is converting into hex and [pack](http://ruby-doc.org/core-2.2.2/Array.html#method-i-pack) appears to be converting to unicode or ASCII, depending on whether it's reading a `\udddd` or `\xdd` input.